Promtail dry-run crashes with external_labels in client config

**Describe the bug**
Promtail crashes if client config has any `external_labels` specified and `-dry-run` option is used.

**To Reproduce**
Consider following clients section in `config.yaml`:
```
clients:
  - url: http://loki.instance:3100/loki/api/v1/push
    external_labels:
      my_label: my
```
Launching promtail in dry-run mode fails:
```
C:\promtail\promtail-windows-amd64-v1.4.1.exe -config.file=C:\promtail\config.yaml -dry-run
Clients configured:
panic: Option ,inline needs a map with string keys in struct flagext.LabelSet [recovered]
        panic: Option ,inline needs a map with string keys in struct flagext.LabelSet

goroutine 1 [running]:
gopkg.in/yaml%2ev2.handleErr(0xc0005ef290)
        /src/loki/vendor/gopkg.in/yaml.v2/yaml.go:249 +0xa1
panic(0x2188ce0, 0xc0000bd840)
        /usr/local/go/src/runtime/panic.go:679 +0x1c0
gopkg.in/yaml%2ev2.(*encoder).structv(0xc00001e2c0, 0x0, 0x0, 0x242cd40, 0xc000106728, 0x99)
        /src/loki/vendor/gopkg.in/yaml.v2/encode.go:211 +0x11f
gopkg.in/yaml%2ev2.(*encoder).marshal(0xc00001e2c0, 0x0, 0x0, 0x242cd40, 0xc000106728, 0x99)
        /src/loki/vendor/gopkg.in/yaml.v2/encode.go:160 +0x767
gopkg.in/yaml%2ev2.(*encoder).structv.func1()
        /src/loki/vendor/gopkg.in/yaml.v2/encode.go:226 +0x55f
gopkg.in/yaml%2ev2.(*encoder).mappingv(0xc00001e2c0, 0x0, 0x0, 0xc0005eecc0)
        /src/loki/vendor/gopkg.in/yaml.v2/encode.go:256 +0x14d
gopkg.in/yaml%2ev2.(*encoder).structv(0xc00001e2c0, 0x0, 0x0, 0x2461300, 0xc000106680, 0x99)
        /src/loki/vendor/gopkg.in/yaml.v2/encode.go:213 +0xfe
gopkg.in/yaml%2ev2.(*encoder).marshal(0xc00001e2c0, 0x0, 0x0, 0x2461300, 0xc000106680, 0x99)
        /src/loki/vendor/gopkg.in/yaml.v2/encode.go:160 +0x767
gopkg.in/yaml%2ev2.(*encoder).marshalDoc(0xc00001e2c0, 0x0, 0x0, 0x2461300, 0xc000106680, 0x99)
        /src/loki/vendor/gopkg.in/yaml.v2/encode.go:93 +0x115
gopkg.in/yaml%2ev2.Marshal(0x2461300, 0xc000106680, 0x0, 0x0, 0x0, 0x0, 0x0)
        /src/loki/vendor/gopkg.in/yaml.v2/yaml.go:203 +0x377
github.com/grafana/loki/pkg/promtail/client.NewLogger(0xc000106410, 0x1, 0x1, 0x1, 0xc0000063f8, 0x30, 0x30)
        /src/loki/pkg/promtail/client/logger.go:45 +0x240
github.com/grafana/loki/pkg/promtail.New(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /src/loki/pkg/promtail/promtail.go:35 +0x178
main.main()
        /src/loki/cmd/promtail/main.go:61 +0x2c8
```
Enclosing into double quotes like this `"my_label": "my"` doesn't make any difference.
Launching promtail without `-dry-run` option works normally and do not crash application.
Removing single line from `config.yaml` fixes the issue:
```
clients:
  - url: http://loki.instance:3100/loki/api/v1/push
    external_labels:
```
**Expected behavior**
Dry-run should work with any valid configuration file.

**Environment:**
promtail-windows-amd64-v1.4.1.exe -version
promtail, version 1.4.1 (branch: HEAD, revision: 2445ccd2)
  build user:       root@bb5e00b6b7f4
  build date:       2020-04-06T20:29:26Z
  go version:       go1.13.4

--- a/pkg/util/flagext/labelset.go
+++ b/pkg/util/flagext/labelset.go
@@ -7,6 +7,9 @@ import (
 	"strings"
 
 	"github.com/prometheus/common/model"
+	"gopkg.in/yaml.v2"
+
+	"github.com/grafana/loki/pkg/util"
 )
 
 // LabelSet is a labelSet that can be used as a flag.
@@ -77,4 +80,13 @@ func (v *LabelSet) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	}
 	v.LabelSet = lbSet
 	return nil
+}
+
+// MarshalYAML implements yaml.Marshaller.
+func (v LabelSet) MarshalYAML() (interface{}, error) {
+	out, err := yaml.Marshal(util.ModelLabelSetToMap(v.LabelSet))
+	if err != nil {
+		return nil, err
+	}
+	return string(out), nil
 }

